semgrep-core/data/basic_tainting.yml

```diff
@@ -0,0 +1,13 @@
+rules:
+  - id: classic
+    pattern-sources:
+      - source(...)
+      - source1(...)
+    pattern-sinks:
+      - sink(...)
+      - eval(...)
```

Comment:
Just to be sure, these are logical OR clauses. In addition to this, is it possible to define two sinks combined by AND clause?

For example, when we were doing path-injection, we had to make patterns like
```
$P = os.path.join(..., request.$W.get(...), ...)
...
open($P, ...)
```
where data had to flow through `os.path.join` as well as `open` to be considered potentially malicious.

Reply:
hmm, right now the tainting by default consider that any tainted argument taint also the return value (unless it's a sanitize function), so here by default $P will be marked as tainted